tests/unit/test_charm.py

```diff
@@ -54,16 +59,124 @@ def test_install_event(update, harness):
     assert isinstance(harness.charm.model.unit.status, ActiveStatus)
 
 
+@patch("charm.KubernetesServicePatch", lambda x, y: None)
+def test_generate_dex_auth_config_raises(harness):
+    """Check the method raises when static login is disabled and no connectors are provided."""
+    harness.begin()
+    config_updates = {
+        "enable-password-db": False,
+        "port": 5555,
+        "public-url": "dummy.url",
+    }
+
+    harness.update_config(config_updates)
+
+    with pytest.raises(CheckFailedError) as error:
+        harness.charm._generate_dex_auth_config()
+    assert (
+        error.value.msg
+        == "Please add a connectors configuration to proceed without a static login."
+    )
+    assert error.value.status_type == BlockedStatus
+
+
+@pytest.mark.parametrize(
+    "dex_config",
+    (
+        {
+            "enable-password-db": False,
+            "port": 5555,
+            "public-url": "dummy.url",
+            "connectors": "test-connector",
+        },
+        {
+            "enable-password-db": True,
+            "port": 5555,
+            "public-url": "dummy.url",
+            "static-username": "new-user",
+            "static-password": "new-pass",
+        },
+    ),
+)
+@patch("charm.Operator._update_layer")
+@patch.object(Operator, "ensure_state", ensure_state)
+@patch("charm.KubernetesServicePatch", lambda x, y: None)
+def test_generate_dex_auth_config_returns(update_layer, dex_config, harness):
+    """Check the method returns dex-auth configuration when different settings are provided."""
+    harness.set_leader(True)
+    harness.begin()
+    harness.set_can_connect("dex", True)
+
+    harness.update_config(dex_config)
+
+    test_configuration = harness.charm._generate_dex_auth_config()
+    assert test_configuration is not None
+
+    # Assert the charm config (generated from dex_config) is the same as the config
+    # generated by calling _generate_dex_auth_config()
+
+    # Load the test_configuration so we can retrieve it as it is passed to the dex-auth service
+    test_configuration_dict = yaml.safe_load(test_configuration)
+
+    # Load the connector configuration to match the generated config in the method
+    assert (
+        yaml.safe_load(harness.model.config["connectors"]) == test_configuration_dict["connectors"]
+    )
+
+    # Check if enablePasswordDB is enabled or not
+    assert (
+        harness.model.config["enable-password-db"] == test_configuration_dict["enablePasswordDB"]
+    )
+
+    # Check if the static username (str) are the same in the charm config and the service config
```

Comment:
Same as above. If not dropped:
```suggestion
    # Check if the static username (str) is the same in the charm config and the service config
```

I'm for dropping BTW, since these comments seem to grow more and more ambiguous (e.g. ref to service config)😇